docs/how-to-guides/convert-changelog-to-table.md

```diff
@@ -10,113 +10,145 @@ keywords: changelog, table, materialize
-
-Begin by telling ksqlDB to start all queries from the earliest point in each topic.
+Begin by telling ksqlDB to start all queries from the earliest point in each topic:
 
 ```sql
 SET 'auto.offset.reset' = 'earliest';
 ```
 
-Make a stream `s1` with four columns. `k` represents the key of the table. Rows with the same key represent information about the same entity. `v1`, `v2`, and `v3` are various value columns.
-
+Make a table `latest_view` with four columns. `k` represents the key of the table. Rows with the same key represent information about the same entity. `v1`, `v2`, and `v3` are various value columns.
 ```sql
-CREATE STREAM s1 (
-    k VARCHAR KEY,
+CREATE SOURCE TABLE latest_view (
+    k VARCHAR PRIMARY KEY,
     v1 INT,
     v2 VARCHAR,
     v3 BOOLEAN
 ) WITH (
-    kafka_topic = 's1',
+    kafka_topic = 'changelog',
     partitions = 1,
-    value_format = 'avro'
+    value_format = 'JSON'
 );
 ```
 
-Insert some rows into `s1`, repeating entries for some of the keys.
+Now, you can view the latest values for each key in your `changelog` topic by issuing a pull query against the `latest_view` table that we just created above:
+```sql
+SELECT * FROM latest_view;
++------+------+------+------+
+|K     |V1    |V2    |V3    |
++------+------+------+------+
+|k1    |2     |c     |false |
+|k2    |4     |e     |true  |
+|k3    |3     |d     |true  |
+```
+
+Notice how for each key, the columns reflect the latest set of values.
 
+If you just want to lookup the latest value for a particular key (let's assume `k2` in this case), then you can simply issue a pull query for that particular key:
 ```sql
-INSERT INTO s1 (
+SELECT * FROM latest WHERE k = 'k2';
++------+------+------+------+
+|K     |V1    |V2    |V3    |
++------+------+------+------+
+|k2    |4     |e     |true  |
+```
+
+
+## Materializing a changelog `STREAM`
+
+Let's say you have a `STREAM` of events in ksqlDB that represent a series of changes called `changelog_stream`, and you want view of the data that reflects the latest values for each key. 
+
+Begin by telling ksqlDB to start all queries from the earliest point in each topic:
+```sql
+SET 'auto.offset.reset' = 'earliest';
+```
+
+Let's mimic adding a few records to the `changelog_stream` by using the `INSERT INTO` statement:
+```sql
+INSERT INTO changelog_stream (
     k, v1, v2, v3
 ) VALUES (
     'k1', 0, 'a', true
 );
 
-INSERT INTO s1 (
+INSERT INTO changelog_stream (
     k, v1, v2, v3
 ) VALUES (
     'k2', 1, 'b', false
 );
 
-INSERT INTO s1 (
+INSERT INTO changelog_stream (
     k, v1, v2, v3
 ) VALUES (
     'k1', 2, 'c', false
 );
 
-INSERT INTO s1 (
+INSERT INTO changelog_stream (
     k, v1, v2, v3
 ) VALUES (
     'k3', 3, 'd', true
 );
 
-INSERT INTO s1 (
+INSERT INTO changelog_stream (
     k, v1, v2, v3
 ) VALUES (
     'k2', 4, 'e', true
 );
 ```
```

Comment:
This setup is missing from the table section. Do you think we need to demonstrate how to create the stream when we don't do the same for the table's input topic?

Reply:
it is missing because we can't insert into source tables, and I did not want to confuse/distract the readers by adding some other way of inserting rows into the table (eg. with using a helper stream on the topic and inserting into the stream)

Reply:
it is possible to directly insert into a stream which is why i included it